### PR TITLE
[nameresolution/consul] resolver caching and service deregister

### PR DESCRIPTION
**Pull request by a-elsheikh**
## Description

Updates to the nameresolution/consul component documentation to reflect the changes made in [components-contrib/pr#3121](https://github.com/dapr/components-contrib/pull/3121) - namely, resolver caching and service deregister

## Issue reference

[components-contrib/issue#934](https://github.com/dapr/components-contrib/issues/934)

### CODE REVIEW

**Comment by a-elsheikh** [comment]
FYI @ItalyPaleAle

**Review by ItalyPaleAle — APPROVED** [review]
Thank you!

File: daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md [hunk 1]
```diff
@@ -25,11 +25,11 @@ spec:
...
+If Consul service registration is managed externally from Dapr you need to ensure that the Dapr-to-Dapr internal gRPC port is added to the service metadata under `DAPR_PORT` (this key is configurable) and that the Consul service name matches the Dapr app Id. You can then omit `selfRegister` from the config above.
 
 ## Behaviour
 
-On `init` the Consul component either validates the connection to the configured (or default) agent or registers the service if configured to do so. The name resolution interface does not cater for an "on shutdown" pattern so consider this when using Dapr to register services to Consul as it does not deregister services.
+On startup the Consul component either validates the connection to the configured (or default) agent or registers the service if configured to do so. Similarly, on shutdown the component will deregister the service if configured to do so.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md]:
```suggestion
On startup the Consul component either validates the connection to the configured (or default) agent or registers the service if configured to do so. Similarly, on shutdown the component deregisters the service if configured to do so.
```

File: daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md [hunk 2]
```diff
@@ -46,8 +46,10 @@ The configuration spec is fixed to v1.3.0 of the Consul API
 | Tags         | N        | `[]string` | Configures any tags to include if/when registering services | `- "dapr"`
 | Meta         | N        | `map[string]string` | Configures any additional metadata to include if/when registering services | `DAPR_METRICS_PORT: "${DAPR_METRICS_PORT}"`
 | DaprPortMetaKey | N     | `string` | The key used for getting the Dapr sidecar port from Consul service metadata during service resolution, it will also be used to set the Dapr sidecar port in metadata during registration. If blank it will default to `DAPR_PORT` | `"DAPR_TO_DAPR_PORT"`
-| SelfRegister | N        | `bool` | Controls if Dapr will register the service to Consul. The name resolution interface does not cater for an "on shutdown" pattern so please consider this if using Dapr to register services to Consul as it will not deregister services. If blank it will default to `false` | `true`
+| SelfRegister | N        | `bool` | Controls if Dapr will register the service to consul on startup. If unset it will default to `false` | `true`
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md]:
```suggestion
| SelfRegister | N        | `bool` | Controls if Dapr registers the service to Consul on startup. If unset, defaults to `false` | `true`
```

File: daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md [hunk 3]
```diff
@@ -46,8 +46,10 @@ The configuration spec is fixed to v1.3.0 of the Consul API
...
+| SelfDeregister | N        | `bool` | Controls if Dapr will deregister the service from consul on shutdown. If unset it will default to `false` | `true`
 | AdvancedRegistration | N | [*api.AgentServiceRegistration](https://pkg.go.dev/github.com/hashicorp/consul/api@v1.3.0#AgentServiceRegistration) | Gives full control of service registration through configuration. If configured the component will ignore any configuration of Checks, Tags, Meta and SelfRegister. | See [sample configs](#sample-configurations)
+| UseCache | N        | `bool` | Configures if Dapr will cache the resolved services in-memory. This is done using consul [blocking queries](https://www.consul.io/api-docs/features/blocking) which can be configured via the QueryOptions configuration. If unset it will default to `false` | `true`
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md]:
```suggestion
| UseCache | N        | `bool` | Configures if Dapr caches the resolved services in-memory. This is done using Consul [blocking queries](https://www.consul.io/api-docs/features/blocking) which can be configured via the QueryOptions configuration. If unset, defaults to `false` | `true`
```

File: daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md [hunk 4]
```diff
@@ -46,8 +46,10 @@ The configuration spec is fixed to v1.3.0 of the Consul API
 | Tags         | N        | `[]string` | Configures any tags to include if/when registering services | `- "dapr"`
 | Meta         | N        | `map[string]string` | Configures any additional metadata to include if/when registering services | `DAPR_METRICS_PORT: "${DAPR_METRICS_PORT}"`
 | DaprPortMetaKey | N     | `string` | The key used for getting the Dapr sidecar port from Consul service metadata during service resolution, it will also be used to set the Dapr sidecar port in metadata during registration. If blank it will default to `DAPR_PORT` | `"DAPR_TO_DAPR_PORT"`
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md]:
```suggestion
| DaprPortMetaKey | N     | `string` | The key used for getting the Dapr sidecar port from Consul service metadata during service resolution, it is also used to set the Dapr sidecar port in metadata during registration. If blank, defaults to `DAPR_PORT` | `"DAPR_TO_DAPR_PORT"`
```

**Review by msfussell — COMMENTED** [review]
Some suggested edits

File: daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md (hunk 4 above)
Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-name-resolution/setup-nr-consul.md]: @a-elsheikh - This should target the v1.12 branch, which is the current branch, yes?

**Comment by a-elsheikh** [comment]
> @a-elsheikh - This should target the v1.12 branch, which is the current branch, yes?

hey @msfussell - as per my conversations with @berndverst the aim is to get [components-contrib/pr#3121](https://github.com/dapr/components-contrib/pull/3121) into v1.13 so this PR is inline with that

**Pull request closed without merging**